unittests/test_policies.py

```diff
@@ -147,6 +147,22 @@ def test_retries_bad_check(self):
         self.assertEqual(max_retries, rt.runtime().current_run)
         self.assertEqual(1, self.runner.stats.num_failures())
 
+    def test_retries_bad_check_early(self):
+        max_retries = 2
+        checks = [BadSetupCheckEarly()]
+        self.runner._max_retries = max_retries
+        self.runner.runall(checks)
+
+        # Ensure that the test was retried #max_retries times and failed.
+        self.assertEqual(1, self.runner.stats.num_cases())
+        self.assertEqual(max_retries, rt.runtime().current_run)
+        self.assertEqual(1, self.runner.stats.num_failures())
+
+        # Ensure that the report succeeds and that the output is as expected.
+        report = self.runner.stats.retry_report()
+        self.assertIn("Test BadSetupCheckEarly was retried 2 time(s) and "
+                      "failed.", report)
```

Comment:
This check is quite restrictive. We will have to update the unit test every time we change either the automatic test naming scheme or message. I think we were getting a stack trace. You could then check that no `Traceback` is found in the output as we do with other similar cases.

Reply:
I think the check is even easier here. I think `retry_report()` was raising an exception. So you may simply call it. If it raises an exception, the unit test will fail.

Reply:
Yes, you simply call. If an unexpected exception is thrown, the unit tests will fail.